From 7dd635e8f7d242054a64717e5ba3333ec1758704 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lisandro=20Dami=C3=A1n=20Nicanor=20P=C3=A9rez=20Meyer?=
 <perezmeyer@gmail.com>
Date: Thu, 22 May 2014 20:19:29 -0300
Subject: [PATCH] Remove -Wcast-align from QMAKE_CXXFLAGS.

x86 doesn't care about alignment, and on all other platforms where it
does something it causes build errors, so instead of removing it on
those platforms just don't enable it at all.

Change-Id: Idfeb387099b28af60ba161b6ca678b7c9df17fe1
---
 mkspecs/features/qt_headersclean.prf | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/mkspecs/features/qt_headersclean.prf b/mkspecs/features/qt_headersclean.prf
index ab0a722..0650adf 100644
--- a/mkspecs/features/qt_headersclean.prf
+++ b/mkspecs/features/qt_headersclean.prf
@@ -13,16 +13,11 @@
 *-g++*: QMAKE_CXXFLAGS += -Woverloaded-virtual -Wshadow -Wundef
 
 # Other nice flags
-*-g++*: QMAKE_CXXFLAGS += -Wnon-virtual-dtor -ansi -Wcast-align -Wchar-subscripts -Wpointer-arith -Wformat-security
+*-g++*: QMAKE_CXXFLAGS += -Wnon-virtual-dtor -ansi -Wchar-subscripts -Wpointer-arith -Wformat-security
 
 # Enable pedantic mode, but accept variadic macros and 'long long' usage.
 *-g++*: QMAKE_CXXFLAGS += -Wno-long-long -Wno-variadic-macros -pedantic-errors
 
-contains(QT_ARCH,arm)|contains(QT_ARCH,mips) {
-    # There are outstanding alignment issues in some container classes.
-    *-g++*:QMAKE_CXXFLAGS -= -Wcast-align
-}
-
 QMAKE_CXXFLAGS += -DQT_NO_CAST_TO_ASCII \
                   -DQT_NO_CAST_FROM_ASCII \
                   -DQT_STRICT_ITERATORS \
-- 
2.0.0.rc2

